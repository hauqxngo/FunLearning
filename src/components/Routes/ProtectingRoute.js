import React, { useContext } from "react";
import { Route, Redirect } from "react-router-dom";
import UserContext from "../../UserContext";

// Prevent currentUser to access routes /signup and /login
const ProtectingRoute = ({ exact, path, children }) => {
  const { currentUser } = useContext(UserContext);

  if (currentUser) {
    return <Redirect to="/" />;
  }

  return (
    <Route exact={exact} path={path}>
      {children}
    </Route>
  );
};

export default ProtectingRoute;
