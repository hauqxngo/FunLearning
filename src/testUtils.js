import React from "react";
import UserContext from "./UserContext";

const testUser = {
  username: "testuser",
  first_name: "Test",
  last_name: "User",
  email: "test@test.com",
  photo_url: null,
};

const UserProvider = ({ children, currentUser = testUser }) => (
  <UserContext.Provider value={{ currentUser }}>
    {children}
  </UserContext.Provider>
);

export { UserProvider };
