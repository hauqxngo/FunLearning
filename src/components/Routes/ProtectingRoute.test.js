import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import { UserProvider } from "../../testUtils";
import ProtectingRoute from "./ProtectingRoute";

it("renders without crashing", () => {
  render(
    <MemoryRouter>
      <UserProvider>
        <ProtectingRoute />
      </UserProvider>
    </MemoryRouter>
  );
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <ProtectingRoute />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider currentUser={null}>
        <ProtectingRoute />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
