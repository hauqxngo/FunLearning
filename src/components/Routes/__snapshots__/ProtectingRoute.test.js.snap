// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `<DocumentFragment />`;

exports[`matches snapshot when logged out 1`] = `<DocumentFragment />`;
